### PR TITLE
Rename micro-gw-default.conf template and Refactor code

### PR DESCRIPTION
**Pull request by AmaliMatharaarachchi**
### Issues

Fixes #https://github.com/wso2/product-microgateway/issues/972
Related to review comments in https://github.com/wso2/product-microgateway/pull/932
Fix removal of api key test in https://github.com/wso2/product-microgateway/pull/934/files#diff-21f65fff21d61dc6ebd2a5076a96bc15L36

### Automation tests
 - Unit tests added: Yes/No
 - Integration tests added: Yes/No

### Tested environments

Not Tested

---
#### Maintainers: Check before merge
- [x] Assigned 'Type' label
- [x] Assigned the project
- [x] Validated respective github issues
- [x] Assigned milestone to the github issue(s)

### CODE REVIEW

File: components/micro-gateway-core/src/main/ballerina/src/gateway/utils/utils.bal [hunk 1]
```diff
@@ -405,6 +405,14 @@ public function printError(string key, string message) {
     log:printError(io:sprintf("[%s] [%s] %s", key, getMessageId(), message));
 }
 
+# Add a full error log with a message and provided key (class) and message ID.
+# + key - The name of the bal file from which the log is printed.
+# + message - The message to be logged.
+# + errorMessage - The error to be logged.
+public function printErrorwithMessage(string key, string message, error errorMessage) {
```
Comment (praminda) [inline components/micro-gateway-core/src/main/ballerina/src/gateway/utils/utils.bal]: Can we use same `printError(string, error)` function with a default-able parameter?

**Comment by praminda** [comment]
I'm not clear about the purpose of the PR. Please add the purpose of the PR and/or any other related PRs to this.

File: components/micro-gateway-core/src/main/ballerina/src/gateway/utils/utils.bal (hunk 1 above)
Comment (AmaliMatharaarachchi) [reply components/micro-gateway-core/src/main/ballerina/src/gateway/utils/utils.bal]: Thanks. Fixed in https://github.com/wso2/product-microgateway/pull/973/commits/f7298983c0031cf9e8587e99c49ff7c4885120c4

**Comment by AmaliMatharaarachchi** [comment]
@praminda updated the PR description

**Pull request merged**